lib/internal/modules/esm/loader.js

```diff
@@ -64,6 +65,11 @@ class Loader {
     if (format === 'builtin')
       return { url: `node:${url}`, format };
 
+    if (this._resolve !== defaultResolve) {
+      // throws ERR_INVALID_URL for resolve if not valid
```

Comment:
I think here's one of the places we can use a more descriptive error message. Something like "invalid URL returned by loader" is preferred over the user having to research why an error is thrown.

Reply:
stack trace should be enough right?

Reply:
I don't think so in this case, because it's not clear what the call hierarchy is. For example, if it's coming from the `URL` constructor, then it's pretty obvious. But no one would know what `Loader.resolve` is without reading the source code.

Reply:
Sure, I've done some polishing on the exact error messages.